Tests/MapboxCoreNavigationTests/MapboxCoreNavigationTests.swift

```diff
@@ -94,23 +117,34 @@ class MapboxCoreNavigationTests: XCTestCase {
     
     func testNewStep() {
         // Coordinates from beginning of step[1] to end of step[2]
-        let coordinates = route.legs[0].steps[1].shape!.coordinates + route.legs[0].steps[2].shape!.coordinates
-        let locations: [CLLocation]
+        let steps = route.legs[0].steps
+        let coordinates = (steps[1].shape?.coordinates ?? []) + (steps[2].shape?.coordinates ?? [])
+        
         let now = Date()
-        locations = coordinates.enumerated().map { CLLocation(coordinate: $0.element,
-                                                              altitude: -1, horizontalAccuracy: -1, verticalAccuracy: -1, course: -1, speed: 10,
-                                                              timestamp: now + $0.offset) }
+        let locations = coordinates.enumerated().map {
+            CLLocation(coordinate: $0.element,
+                       altitude: -1,
+                       horizontalAccuracy: -1,
+                       verticalAccuracy: -1,
+                       timestamp: now + $0.offset)
+        }
         
-        navigation = MapboxNavigationService(route: route, routeIndex: 0, routeOptions: routeOptions, directions: directions, simulating: .never)
+        navigation = MapboxNavigationService(route: route,
+                                             routeIndex: 0,
+                                             routeOptions: routeOptions,
+                                             directions: directions,
+                                             simulating: .never)
         expectation(forNotification: .routeControllerDidPassSpokenInstructionPoint, object: navigation.router) { (notification) -> Bool in
-            let routeProgress = notification.userInfo![RouteController.NotificationUserInfoKey.routeProgressKey] as? RouteProgress
+            let routeProgress = notification.userInfo?[RouteController.NotificationUserInfoKey.routeProgressKey] as? RouteProgress
             
             return routeProgress?.currentLegProgress.stepIndex == 2
         }
         
         navigation.start()
         
-        locations.forEach { navigation.router!.locationManager!(navigation.locationManager, didUpdateLocations: [$0])}
+        locations.forEach {
+            navigation.router?.locationManager?(navigation.locationManager, didUpdateLocations: [$0])
+        }
```

Comment:
Tests are consistently failing, even though this PR only reformats this method:

```
Test Case '-[MapboxCoreNavigationTests.MapboxCoreNavigationTests testNewStep]' started.
2021-06-10 17:11:55.083155+0000 xctest[3367:66941] This app has attempted to access privacy-sensitive data without a usage description. The app's Info.plist must contain an “NSLocationWhenInUseUsageDescription” key with a string value explaining to the user how the app uses this data
lanes: nil
lanes: nil
lanes: nil
lanes: nil
2021-06-10 17:11:55.085502+0000 xctest[3367:67542] <nav-native> No PersistentConfig found at navigation/config.json
lanes: nil
lanes: nil
2021-06-10 17:11:55.085882+0000 xctest[3367:67542] <nav-native> Unable to find the latest version of routing tiles!
lanes: nil
lanes: nil
lanes: nil
lanes: nil
lanes: nil
2021-06-10 17:11:55.086291+0000 xctest[3367:67542] <nav-native> Async config init - Unable to determine the routing tiles version
lanes: nil
…
lanes: nil
2021-06-10 17:11:55.097732+0000 xctest[3367:66941] Warning: <nav-native> Location at timestamp 1623345115081732096ns was filtered out.
2021-06-10 17:11:55.098354+0000 xctest[3367:67575] Warning: <nav-native> Got location timestamp from the past: 1623345116081732096ns <= 1623345116095142912ns. No status will be produced.
/Users/distiller/project/Tests/MapboxCoreNavigationTests/MapboxCoreNavigationTests.swift:149: error: -[MapboxCoreNavigationTests.MapboxCoreNavigationTests testNewStep] : Asynchronous wait failed: Exceeded timeout of 5 seconds, with unfulfilled expectations: "Expect notification 'RouteControllerDidPassSpokenInstructionPoint' from <MapboxCoreNavigation.RouteController: 0x7facf80ce1e0>".
/Users/distiller/project/Tests/MapboxCoreNavigationTests/MapboxCoreNavigationTests.swift:150: error: -[MapboxCoreNavigationTests.MapboxCoreNavigationTests testNewStep] : XCTAssertNil failed: "Error Domain=com.apple.XCTestErrorDomain Code=0 "(null)""
2021-06-10 17:12:00.568802+0000 xctest[3367:66941] <nav-native> Path to history directory couldn't be empty
2021-06-10 17:12:00.569621+0000 xctest[3367:66941] <nav-native> PersistentConfig path: navigation/config.json
2021-06-10 17:12:00.569904+0000 xctest[3367:66941] <nav-native> Tiles path = ""
2021-06-10 17:12:00.570175+0000 xctest[3367:66941] <nav-native> TileStore instance was passed
2021-06-10 17:12:00.570438+0000 xctest[3367:66941] <nav-native> Endpoint config = "{"host":"https://example.com","isFallback":false,"minDiffInDaysToConsiderServerVersion":0,"navigatorVersion":"52.0.0","token":"***...*","userAgent":"MapboxCoreNavigation/2.0.0 MapboxDirections/2.0.0 iOS/14.4.0 (x86_64)","version":"","versionBeforeFallback":""}"
2021-06-10 17:12:00.570768+0000 xctest[3367:66941] <nav-native> Creating navigator v.52.0.0
2021-06-10 17:12:00.576749+0000 xctest[3367:67543] <nav-native> No PersistentConfig found at navigation/config.json
2021-06-10 17:12:00.605907+0000 xctest[3367:67543] <nav-native> Unable to find the latest version of routing tiles!
2021-06-10 17:12:00.606070+0000 xctest[3367:67543] <nav-native> Async config init - Unable to determine the routing tiles version
Test Case '-[MapboxCoreNavigationTests.MapboxCoreNavigationTests testNewStep]' failed (5.528 seconds).
```

Maybe the issue is that these lines are feeding the navigator locations in a tight loop without waiting for a callback indicating that the navigator has finished responding to each location. If the navigator expects to get location updates in series, then ideally it would also have a method that accepts an array of these batched locations, so that we don’t have to pump them into the navigator in a for loop.

As it is, it would be inconvenient but possible to refactor this test to wait for a progress change notification before feeding the navigator an additional location. However, that would not accurately exercise what happens in the wild when Core Location gives RouteController a batch of locations all at once. Essentially, MapboxNavigationNative should support the same semantics as `CLLocationManagerDelegate.locationManager(_:didUpdateLocations:)` as opposed to `CLLocationManagerDelegate.locationManager(_:didUpdateTo:from:)`, which was deprecated in iOS 6.0.

/cc @mapbox/navnative